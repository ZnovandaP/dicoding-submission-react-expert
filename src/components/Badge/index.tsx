import * as React from 'react';

type BadgeProps = {
  children: React.ReactNode
};

export default function Badge({ children }: BadgeProps) {
  return (
    <div className="center text-xs px-4 py-1 ring-1 ring-primary rounded-full">
      {children}
    </div>
  );
}
