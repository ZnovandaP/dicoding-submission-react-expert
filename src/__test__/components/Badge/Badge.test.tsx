/* eslint-disable react/react-in-jsx-scope */

/*
Skenario test component Badge:
* should render correctly component Badge
* should render component Badge with size 'sm'
* should render component Badge with size 'md'
*/

import Badge from '@/components/Badge';
import {
  describe, it, expect,
} from '@jest/globals';
import { render, screen } from '@testing-library/react';

describe('test Badge component', () => {
  it('should render correctly component Badge', () => {
    render(
      <Badge>
        default
      </Badge>,
    );

    const badge = screen.getByText(/default/);

    expect(badge).toMatchSnapshot();
  });

  it('should render component Badge with size sm', () => {
    render(
      <Badge size="sm">
        smallBadge
      </Badge>,
    );

    const badge = screen.getByText(/smallBadge/);

    expect(badge.classList.contains('text-xs')).toBeTruthy();
  });

  it('should render component Badge with size md', () => {
    render(
      <Badge size="md">
        mediumBadge
      </Badge>,
    );

    const badge = screen.getByText(/mediumBadge/);

    expect(badge.classList.contains('text-base')).toBeTruthy();
  });
});
