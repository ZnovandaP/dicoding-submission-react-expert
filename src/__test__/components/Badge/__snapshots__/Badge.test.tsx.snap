// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Badge component should render correctly component Badge 1`] = `
<div
  class="center text-xs px-4 py-1 ring-1 ring-primary rounded-full"
>
  default
</div>
`;
